### DISCUSSION TITLE
Is there a way to update value on SlateEditor by using useAllFormFields with dispatch?

### DISCUSSION THREAD

**Discussion by xxxle0** [payloadcms/payload#5167] [Q&A]
I'm trying to update the value from slateEditor from another component event. I find out that I can update value of some simple component like Checkbox, Text, TextArea with `dispatch` Update but not for SlateEditor. Any suggestions on it?

**Comment by mvdve** [comment]
I got this working with the useFormsFields hook but this should also work with useAllFormFields. Type definition is just a simple mockup for now.

```ts
type TextNode = [{[x: string]: unknown; text: string; children?: TextNode; type?: string}];

const [value, setValue]: [TextNode, (value: TextNode) => void] = useFormFields(([fields, dispatch]) => [
    fields[path].initialValue as TextNode,
    (value: TextNode) => dispatch({type: "UPDATE", path: path, initialValue: value}),
]);
```

With string fields, you can use the value property, with the richText field (SlateJS editor), take the initialValue property. Otherwise the editor content won't rerender.
